### PR TITLE
Residential Call Forward services

### PR DESCRIPTION
**Pull request by Kaian**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [ ] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 

#### Description
Added new Brand Services to enable or disable Residential Devices Call Forward Settings

This PR requires the sounds available at https://github.com/irontec/asterisk-sounds/commit/0ed90250e4751fffe8ab1db99eb6b23ec5f996e3

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Service/CompanyService/PropagateBrandServices.php [hunk 1]
```diff
@@ -55,13 +56,31 @@ public function execute(CompanyInterface $company)
             $company->getBrand()->getId()
         );
 
+        $availableServices = [
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Service/CompanyService/PropagateBrandServices.php]: What about AVAILABLE_SERVICES constant on Service itself?

File: web/admin/application/library/IvozProvider/Klear/Filter/CompanyServices.php [hunk 2]
```diff
@@ -47,9 +48,22 @@ public function setRouteDispatcher(KlearMatrix_Model_RouteDispatcher $routeDispa
             ["BrandService.brand = " . $currentBrandyId]
         );
 
+        $availableServices = [
```
Comment (mmadariaga) [inline web/admin/application/library/IvozProvider/Klear/Filter/CompanyServices.php]: What about AVAILABLE_SERVICES constant on Service itself?

File: web/admin/application/library/IvozProvider/Klear/Filter/CompanyServices.php [hunk 3]
```diff
@@ -47,9 +48,22 @@ public function setRouteDispatcher(KlearMatrix_Model_RouteDispatcher $routeDispa
...
+            Service::CLOSE_LOCK,
+            Service::OPEN_LOCK,
+            Service::TOGGLE_LOCK,
+        ];
+
         $brandServicesIds = array();
         foreach ($brandServices as $brandService) {
-            array_push($brandServicesIds, $brandService->getServiceId());
+            $serviceIden = $brandService->getService()->getIden();
+            if (in_array($serviceIden, $availableServices)) {
```
Comment (mmadariaga) [inline web/admin/application/library/IvozProvider/Klear/Filter/CompanyServices.php]: in_array causes unexpected results without $strict flag on, we should use it by default => in_array ( mixed $needle , array $haystack , true)

File: asterisk/agi/src/Agi/Action/ServiceAction.php [hunk 4]
```diff
@@ -379,6 +397,101 @@ protected function processToggleLock()
...
+        $callForwardSetting = array_shift($callForwardSettings);
+
+        if (empty($destination)) {
+            // Remove existing CallForward
+            if ($callForwardSetting) {
+                $this->entityTools->remove($callForwardSetting);
+            }
+            // Feedback sound
+            $this->agi->notice("%s CFW disabled for %s", $callForwardType, $caller);
+            $this->agi->playback("ivozprovider/cfw-disabled");
```
Comment (mmadariaga) [inline asterisk/agi/src/Agi/Action/ServiceAction.php]: What about a guard clause here? return + no else condition

File: asterisk/agi/src/Agi/Agents/ResidentialAgent.php [hunk 5]
```diff
@@ -49,6 +50,12 @@ public function getLanguageCode()
         return $this->residential->getLanguageCode();
     }
 
+    public function getCallForwardSettings(\Doctrine\Common\Collections\Criteria $criteria = null)
```
Comment (mmadariaga) [inline asterisk/agi/src/Agi/Agents/ResidentialAgent.php]: I would move Criteria fqdn to "use" statements at the top

**Comment by ironArt3mis** [comment]
Rebase completed.

**Pull request merged**